### PR TITLE
implementing fragment navigation

### PR DESCRIPTION
**Pull request by benjaglobant**
In this PR I have implemented the communication between activity and fragments, using the menu items to move to each fragment.
Gif showing the behavior of the UI (recorded in a Nexus S):
firstPR

### CODE REVIEW

File: app/src/main/java/com/globant/marvelmvvm/view/activity/MainActivity.kt [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+                Toast.makeText(this, getString(R.string.string_specific_character_message), Toast.LENGTH_SHORT).show()
+                navController.navigate(R.id.specificCharacterFragment)
+            }
+        }
+        drawer_layout.closeDrawer(GravityCompat.START)
+        return true
+    }
+
+    companion object {
+        const val ALL_CHARACTERS = "All Characters"
```
Comment (lucianoeli) [inline app/src/main/java/com/globant/marvelmvvm/view/activity/MainActivity.kt]: add these ones to string resources

File: app/src/main/res/layout/activity_main.xml [hunk 2]
```diff
@@ -1,18 +1,49 @@
...
+            android:id="@+id/fragment"
+            android:name="androidx.navigation.fragment.NavHostFragment"
+            android:layout_width="match_parent"
+            android:layout_height="match_parent"
+            app:defaultNavHost="true"
+            app:navGraph="@navigation/nav_graph" />
+
+    </LinearLayout>
+
+    <com.google.android.material.navigation.NavigationView
```
Comment (SartiMau) [inline app/src/main/res/layout/activity_main.xml:35]: maybe you can use a style here

**Review by SartiMau — APPROVED** [review]

File: app/src/main/java/com/globant/marvelmvvm/view/activity/MainActivity.kt (hunk 1 above)
Comment (SartiMau) [reply app/src/main/java/com/globant/marvelmvvm/view/activity/MainActivity.kt]: +1

**Review by SartiMau — CHANGES_REQUESTED** [review]
+1 after comments

File: app/src/main/res/layout/activity_main.xml [hunk 3]
```diff
@@ -1,18 +1,49 @@
...
+            android:id="@+id/toolbar"
+            android:layout_width="match_parent"
+            android:layout_height="wrap_content"
+            android:layout_margin="@dimen/margin_toolbar"
+            android:background="@drawable/toolbar_background"
+            app:title="@string/string_toolbar_title"
+            app:titleTextAppearance="@style/style_navigation_view_item" />
+
+        <fragment
+            android:id="@+id/fragment"
```
Comment (nicolbidone) [inline app/src/main/res/layout/activity_main.xml]: pls use more descriptive names

File: app/src/main/res/layout/activity_main.xml [hunk 4]
```diff
@@ -1,18 +1,49 @@
...
+    tools:openDrawer="start">
 
-    <TextView
+    <LinearLayout
+        android:layout_width="match_parent"
+        android:layout_height="match_parent"
+        android:orientation="vertical">
+
+        <androidx.appcompat.widget.Toolbar
+            android:id="@+id/toolbar"
```
Comment (nicolbidone) [inline app/src/main/res/layout/activity_main.xml]: pls use more descriptive names

File: app/src/main/res/layout/fragment_all_characters.xml [hunk 5]
```diff
@@ -0,0 +1,14 @@
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".view.fragment.AllCharactersFragment"
+    android:background="@android:color/black">
+
+    <ImageView
+        android:id="@+id/background_image"
```
Comment (nicolbidone) [inline app/src/main/res/layout/fragment_all_characters.xml]: pls use more descriptive names

File: app/src/main/res/layout/menu_header.xml [hunk 6]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0" encoding="utf-8"?>
+<ImageView xmlns:android="http://schemas.android.com/apk/res/android"
+    android:id="@+id/imageView"
```
Comment (nicolbidone) [inline app/src/main/res/layout/menu_header.xml]: pls use more descriptive names

File: app/src/main/res/layout/activity_main.xml [hunk 7]
```diff
@@ -1,18 +1,49 @@
-<?xml version="1.0" encoding="utf-8"?>
-<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<androidx.drawerlayout.widget.DrawerLayout xmlns:android="http://schemas.android.com/apk/res/android"
     xmlns:app="http://schemas.android.com/apk/res-auto"
     xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/drawer_layout"
```
Comment (nicolbidone) [inline app/src/main/res/layout/activity_main.xml]: pls use more descriptive names

File: app/src/main/java/com/globant/marvelmvvm/view/activity/MainActivity.kt (hunk 1 above)
Comment (benjaglobant) [reply app/src/main/java/com/globant/marvelmvvm/view/activity/MainActivity.kt]: Done!

File: app/src/main/res/layout/activity_main.xml (hunk 2 above)
Comment (benjaglobant) [reply app/src/main/res/layout/activity_main.xml:35]: Done!

File: app/src/main/res/layout/activity_main.xml (hunk 3 above)
Comment (benjaglobant) [reply app/src/main/res/layout/activity_main.xml]: Done!

File: app/src/main/res/layout/fragment_all_characters.xml (hunk 5 above)
Comment (benjaglobant) [reply app/src/main/res/layout/fragment_all_characters.xml]: Done!

File: app/src/main/res/layout/activity_main.xml (hunk 4 above)
Comment (benjaglobant) [reply app/src/main/res/layout/activity_main.xml]: Done!

File: app/src/main/res/layout/menu_header.xml (hunk 6 above)
Comment (benjaglobant) [reply app/src/main/res/layout/menu_header.xml]: Done!

File: app/src/main/res/layout/activity_main.xml (hunk 7 above)
Comment (benjaglobant) [reply app/src/main/res/layout/activity_main.xml]: Done!

**Review by SartiMau — APPROVED** [review]

**Review by lucianoeli — APPROVED** [review]

File: app/src/main/java/com/globant/marvelmvvm/view/activity/MainActivity.kt [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        when (item.itemId) {
+            R.id.allCharactersFragment -> {
+                activity_main_toolbar.title = getString(R.string.string_all_characters)
+                Toast.makeText(this, getString(R.string.string_all_characters_message), Toast.LENGTH_SHORT).show()
+                navController.navigate(R.id.allCharactersFragment)
+            }
+            R.id.specificCharacterFragment -> {
+                activity_main_toolbar.title = getString(R.string.string_specific_character)
+                Toast.makeText(this, getString(R.string.string_specific_character_message), Toast.LENGTH_SHORT).show()
+                navController.navigate(R.id.specificCharacterFragment)
```
Comment (mauroLonghi) [inline app/src/main/java/com/globant/marvelmvvm/view/activity/MainActivity.kt]: Look than in both cases repeat the methods with different s "values", so you can create a method and gives the different value in each case

Comment (benjaglobant) [reply app/src/main/java/com/globant/marvelmvvm/view/activity/MainActivity.kt]: Done!

**Review by lucianoeli — APPROVED** [review]

**Review by mauroLonghi — APPROVED** [review]

**Review by SartiMau — APPROVED** [review]

**Review by nicolbidone — APPROVED** [review]

**Pull request merged**